runtime/src/bank.rs

```diff
@@ -1210,16 +1211,66 @@ impl Bank {
         }
     }
 
+    fn distribute_rent_to_validators(
+        &self,
+        vote_account_hashmap: &HashMap<Pubkey, (u64, Account)>,
+        rent_to_be_distributed: u64,
+    ) {
+        let mut node_stake_hashmap: HashMap<Pubkey, u64> = HashMap::new();
+        let mut total_staked = 0;
+
+        vote_account_hashmap
+            .iter()
+            .map(|(_vote_pubkey, (staked, account))| {
+                total_staked += *staked;
+                VoteState::deserialize(&account.data)
+                    .ok()
+                    .map(|vote_state| (*staked, vote_state.node_pubkey))
```

Comment:
could filter_map()

Reply:
should filter_map(), we don't really want to pay Pubkey::default()

Reply:
updated